diracx-db/src/diracx/db/sql/utils/job_status.py

```diff
@@ -16,11 +16,47 @@
 from .. import JobDB, JobLoggingDB, SandboxMetadataDB, TaskQueueDB
 
 
+async def set_job_statuses(
+    job_update: dict[int, dict[datetime, JobStatusUpdate]],
+    config: Config,
+    job_db: JobDB,
+    job_logging_db: JobLoggingDB,
+    task_queue_db: TaskQueueDB,
+    background_task: BackgroundTasks,
+    force: bool = False,
+):
+    """Bulk operation setting status on multiple job IDs, returning a dictionary of job ID to result.
+    This is done by calling set_job_status for each ID and status dictionary provided within a ForgivingTaskGroup.
+
+    """
+    async with ForgivingTaskGroup() as tg:
+        results = [
+            tg.create_task(
+                set_job_status(
+                    job_id,
+                    status_dict,
+                    config,
+                    job_db,
+                    job_logging_db,
+                    task_queue_db,
+                    background_task,
+                    force=force,
+                )
+            )
+            for job_id, status_dict in job_update.items()
+        ]
+
+    return {job_id: status for job_id, status in zip(job_update.keys(), results)}
```

Comment:
I think you have to call `.result()` on the tasks?

I think this would be clearer:

```suggestion
    async with ForgivingTaskGroup() as tg:
        tasks = {
            job_id: tg.create_task(
                set_job_status(
                    job_id,
                    status_dict,
                    config,
                    job_db,
                    job_logging_db,
                    task_queue_db,
                    background_task,
                    force=force,
                )
            )
            for job_id, status_dict in job_update.items()
        }

    return {k: v.result() for k, v in tasks.items()}
```